### ISSUE TITLE
twisted.test.test_logfile.testNoPermission failing will break buildbot

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2044]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)| @jerub reported|
|-|-|
|Trac ID|trac#2044|
|Type|defect|
|Created|2006-09-01 01:18:54Z|

twisted.test.test_logfile.testNoPermission does a chmod 444 of a directory. this should be a 0555 instead.

Also, there should be a try/finally so if the test fails, it cleans up after itself properly.

<details><summary>Searchable metadata</summary>

```
trac-id__2044 2044
type__defect defect
reporter__jerub jerub
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1157073534000000 1157073534000000
changetime__1157266400000000 1157266400000000
version__None None
owner__jerub jerub

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_1"></a>|@jerub commented|
|-|-|

I decided to leave this as 0444. 

This passes tests on python2.5 and the windows slaves, so I'm marking it ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

Looks good.  Merge.

**Comment by twisted-trac** [comment]
|<img alt="stephen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|stephen set status to `closed`|
|-|-|

(In [18028]) 
Use a try/finally around setting the permissions of a directory during tests. Use an octal literal instead of decimal literal.

Author: Jerub
Reviewer: Exarkun
Fixes [#2044](https://github.com/twisted/twisted/issues/2044)

During tests on the pypy builder where this test (twisted.test.test_logfile.testNoPermission) would traceback, the permissions were never reset, causing the next buildslave build to fail because the directories contents could not be read or removed without chmods, or a root rm -rf.
